sdk/tools/azure-openrewrite/src/main/resources/META-INF/rewrite/rewrite.yml

```diff
@@ -0,0 +1,255 @@
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: com.azure.core.http.policy.KeyCredentialPolicy
+      newFullyQualifiedTypeName: io.clientcore.core.http.pipeline.KeyCredentialPolicy
+
+  # Recipe that changes all instances of com.azure.core.credential.KeyCredential
+  # to io.clientcore.core.credential.KeyCredential
+  # Before:
+  # import com.azure.core.credential.KeyCredential;
+  # After:
+  # import io.clientcore.core.credential.KeyCredential;
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: com.azure.core.credential.KeyCredential
+      newFullyQualifiedTypeName: io.clientcore.core.credential.KeyCredential
+
+  # Recipe that changes all instances of com.azure.core.client.traits.KeyCredentialTrait
+  # to io.clientcore.core.models.traits.KeyCredentialTrait
+  # Before:
+  # import com.azure.core.client.traits.KeyCredentialTrait;
+  # After:
+  # import io.clientcore.core.models.traits.KeyCredentialTrait;
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: com.azure.core.client.traits.KeyCredentialTrait
+      newFullyQualifiedTypeName: io.clientcore.core.models.traits.KeyCredentialTrait
+
+  # NOTE:
+  #    A copy of the client-core jar file must be in the resources\classpath
+  #    directory to reliably parse from clientcore library
+
+  # --------------------------------------------------------------------
+  # Recipes to migrate TextTranslationClientBuilder HttpTrait implemented
+  # methods and parameters
+  # --------------------------------------------------------------------
+  # Change Types from the azure-core to clientcore versions
+  # Before:
+  # import com.azure.core.http.HttpClient;
+  # After:
+  # import io.clientcore.core.http.client.HttpClient;
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: com.azure.core.http.HttpClient
+      newFullyQualifiedTypeName: io.clientcore.core.http.client.HttpClient
+      ignoreDefinition: false
+  # Before:
+  # import com.azure.core.http.HttpPipeline;
+  # After:
+  # import io.clientcore.core.http.pipeline.HttpPipeline;
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: com.azure.core.http.HttpPipeline
+      newFullyQualifiedTypeName: io.clientcore.core.http.pipeline.HttpPipeline
+      ignoreDefinition: false
+  # Before:
+  # import com.azure.core.http.policy.HttpPipelinePolicy;
+  # After:
+  # import io.clientcore.core.http.pipeline.HttpPipelinePolicy;
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: com.azure.core.http.policy.HttpPipelinePolicy
+      newFullyQualifiedTypeName: io.clientcore.core.http.pipeline.HttpPipelinePolicy
+      ignoreDefinition: false
+
+  # Rename methods
+  # Alternatively, can target HttpTrait directly
+
+  # Before:
+  # textTranslationClient.retryOptions(new RetryOptions())
+  # After:
+  # textTranslationClient.httpRetryOptions(new RetryOptions())
+  - org.openrewrite.java.ChangeMethodName:
```

Comment:
Does this show that all OpenRewrite operations need to be in the same `rewrite.yml` where you have to migrate everything at once rather than being able to do piecemeal migrations?

Reply:
@alzimmermsft Correct, it will migrate everything currently supported via OpenRewrite recipes at once.